File: plugin/src/main/java/com/jtprince/util/ChatUtils.java
package com.jtprince.util;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.TextComponent;

import java.util.Collection;
import java.util.Iterator;

public class ChatUtils {
    public static TextComponent commaSeparated(Collection<TextComponent> items) {
        TextComponent.Builder builder = Component.text();

        Iterator<TextComponent> iter = items.iterator();
        while (iter.hasNext()) {
            TextComponent current = iter.next();
            builder.append(current);
            if (iter.hasNext()) {
                builder.append(Component.text(", "));
            }
        }

        return builder.build();
    }
}

File: bukkit/src/main/kotlin/com/jtprince/util/ChatUtils.kt
package com.jtprince.util

import net.kyori.adventure.text.Component
import net.kyori.adventure.text.TextComponent

object ChatUtils {
    fun commaSeparated(items: Collection<Component>): TextComponent {
        val builder = Component.text()
        val iter = items.iterator()
        while (iter.hasNext()) {
            val current = iter.next()
            builder.append(current)
            if (iter.hasNext()) {
                builder.append(Component.text(", "))
            }
        }
        return builder.build()
    }
}
